### PR TITLE
feat(provider): streamflix

### PR DESCRIPTION
**Pull request by lem6ns**
tested tv and movies, subtitles work on both
glorified 2embed

### CODE REVIEW

File: src/backend/providers/streamflix.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      const seasonNumber = media.meta.seasonData.number;
+      const episodeNumber = media.meta.seasonData.episodes.find(
+        (e) => e.id === episode
+      )?.number;
+
+      const streamRes = await proxiedFetch<any>(
+        `/api/player/tv?id=${media.tmdbId}&s=${seasonNumber}&e=${episodeNumber}`,
+        {
+          baseURL: streamflixBase,
+        }
```
Comment (thehairy) [inline src/backend/providers/streamflix.ts:43]:
```suggestion
      const streamRes = await proxiedFetch<any>(
        `/api/player/tv`,
        {
          baseURL: streamflixBase,
          params: {
              id: media.tmdbId,
              s: seasonNumber,
              e: episodeNumber
          },
        }
```

You can also just set the seasonNumber and episodeNumber to 1 if its a movie, or keep it undefined.
Then you dont need such a huge if/else with basically the same code repeating

**Review by thehairy — CHANGES_REQUESTED** [review]

File: src/backend/providers/streamflix.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+  async scrape({ media, episode, progress }) {
+    if (!this.type.includes(media.meta.type)) {
+      throw new Error("Unsupported type");
+    }
+
+    progress(30);
+    const type = media.meta.type === MWMediaType.MOVIE ? "movies" : "tv";
+    let seasonNumber: any;
+    let episodeNumber: any;
```
Comment (thehairy) [inline src/backend/providers/streamflix.ts:35]:
```suggestion
    let seasonNumber: string | undefined;
    let episodeNumber: string | undefined;
```

any just hurts my soul

File: src/backend/providers/streamflix.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+    const streamRes = await proxiedFetch<any>(`/api/player/${type}`, {
+      baseURL: streamflixBase,
+      params: {
+        id: media.tmdbId,
+        s: seasonNumber,
+        e: episodeNumber,
+      },
+    });
+    if (!streamRes.headers.Referer) throw new Error("No watchable item found");
```
Comment (thehairy) [inline src/backend/providers/streamflix.ts:53]: Are you sure thats a reliable way to tell if there is a video?
Haven't looked that deep into the provider.

**Review by thehairy — CHANGES_REQUESTED** [review]

File: src/backend/providers/streamflix.ts (hunk 3 above)
Comment (lem6ns) [reply src/backend/providers/streamflix.ts:53]: > Are you sure thats a reliable way to tell if there is a stream?
> Haven't looked that deep into the provider.

YES! If referer is blank, there is no streams/captions

File: src/backend/providers/streamflix.ts (hunk 2 above)
Comment (lem6ns) [reply src/backend/providers/streamflix.ts:35]: Lol

**Review by thehairy — APPROVED** [review]

**Review by thehairy — APPROVED** [review]

**Review by mrjvs — APPROVED** [review]

**Pull request merged**